### PR TITLE
fix(CON-1104),refactor: allow editing of supply to unlimited/unlimited and minted claims; change name to serendipity

### PR DESCRIPTION
**Pull request by jaxonL**
## This PR
issue: when `totalMax` = 0 (unlimited claim) and the minted count (`total`) was > 0, any edits to the claim (even if we weren't touching the `totalMax` count) would be rejected, as the `totalMax` of 0 was less than the `total` count.

fix: check if `totalMax != 0` before checking the `total` against the `totalMax`.

refactor:
- split out some tests cases for clarity (move deprecate function cases into its own tests; add limited, unlimited supply edit (sanity))
- rename contract to reference serendipity

### Ticket

https://linear.app/manifoldxyz/issue/CON-1104

## CR Notes

main code change, apart from renaming the contract, is this line: https://github.com/manifoldxyz/creator-core-extensions-solidity/pull/96#discussion_r1657889256

## QA Steps

* [ ]

### CODE REVIEW

File: packages/manifold/contracts/gachaclaims/ERC1155SerendipityLazyClaim.sol [hunk 1]
```diff
@@ -105,13 +105,13 @@ contract ERC1155GachaLazyClaim is IERC165, IERC1155GachaLazyClaim, ICreatorExten
...
-    if (instanceId == 0 || instanceId > MAX_UINT_56) revert IGachaLazyClaim.InvalidInstance();
+    if (instanceId == 0 || instanceId > MAX_UINT_56) revert ISerendipityLazyClaim.InvalidInstance();
     if (updateClaimParameters.endDate != 0 && updateClaimParameters.startDate >= updateClaimParameters.endDate)
-      revert IGachaLazyClaim.InvalidDate();
-    if (updateClaimParameters.totalMax < claim.total) revert IGachaLazyClaim.CannotLowerTotalMaxBeyondTotal();
-    if (updateClaimParameters.totalMax > MAX_UINT_32) revert IGachaLazyClaim.InvalidInput();
-    if (updateClaimParameters.storageProtocol == StorageProtocol.INVALID) revert IGachaLazyClaim.InvalidStorageProtocol();
-    if (updateClaimParameters.cost > MAX_UINT_96) revert IGachaLazyClaim.InvalidInput();
+      revert ISerendipityLazyClaim.InvalidDate();
+    if (updateClaimParameters.totalMax != 0 && updateClaimParameters.totalMax < claim.total) revert ISerendipityLazyClaim.CannotLowerTotalMaxBeyondTotal();
```
Comment (jaxonL) [inline packages/manifold/contracts/gachaclaims/ERC1155SerendipityLazyClaim.sol:111]: main change here!!

File: packages/manifold/contracts/gachaclaims/IERC1155SerendipityLazyClaim.sol
Comment (jaxonL) [inline packages/manifold/contracts/gachaclaims/IERC1155SerendipityLazyClaim.sol:1]: changes = renaming

File: packages/manifold/contracts/gachaclaims/ISerendipityLazyClaim.sol [hunk 2]
```diff
@@ -39,9 +39,9 @@ interface IGachaLazyClaim {
   error CannotMintMoreThanReserved();
   error CannotMintFromContract();
 
-  event GachaClaimInitialized(address indexed creatorContract, uint256 indexed instanceId, address initializer);
-  event GachaClaimUpdated(address indexed creatorContract, uint256 indexed instanceId);
-  event GachaClaimMintReserved(
+  event SerendipityClaimInitialized(address indexed creatorContract, uint256 indexed instanceId, address initializer);
+  event SerendipityClaimUpdated(address indexed creatorContract, uint256 indexed instanceId);
+  event SerendipityClaimMintReserved(
```
Comment (jaxonL) [inline packages/manifold/contracts/gachaclaims/ISerendipityLazyClaim.sol:44]: renaming these events -- will need to update scanner + lambda code (possibly) with new events to listen to once deployed

File: packages/manifold/test/gacha/ERC1155GachaLazyClaim.t.sol
Comment (jaxonL) [inline packages/manifold/test/gacha/ERC1155GachaLazyClaim.t.sol:1]: actually just renamed, but i guess the diff was too big 😓

File: packages/manifold/test/gacha/ERC1155SerendipityLazyClaim.t.sol
Comment (jaxonL) [inline packages/manifold/test/gacha/ERC1155SerendipityLazyClaim.t.sol:1]: main changes come from adding tests for the total supply = 0 edge case + moving deprecated tests into its own test cases

**Comment by wwhchung** [comment]
Why don’t you just call it Serendipity and drop LazyClaim?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -104,7 +104,7 @@ jobs:
       - name: Setup LCOV
         uses: hrishikesh-kadam/setup-lcov@v1
       - name: Report code coverage
-        uses: zgosalvez/github-actions-report-lcov@v3
+        uses: zgosalvez/github-actions-report-lcov@v4
```
Comment (jaxonL) [inline .github/workflows/ci.yml:107]: bumping to v4 that bumps to node 20

File: packages/manifold/test/gacha/ERC1155GachaLazyClaim.t.sol
Comment (jaxonL) [inline packages/manifold/test/gacha/ERC1155GachaLazyClaim.t.sol:1]: renamed; diff is too big to register as a same file.

see below for changes.

File: packages/manifold/test/gacha/ERC1155Serendipity.t.sol
Comment (jaxonL) [inline packages/manifold/test/gacha/ERC1155Serendipity.t.sol:1]: namely: refactor update tests to have different cases for limited, unlimited supply; adjust names/references; and move out `deprecate` behaviour tests into its own set of tests
